Fix `Title.select_title_place`

Halbaroth opened the pull request:
The filter function within the `Title.select_title_place` select function had a bug. This commit restores the previous behavior. 

Fix #2287

a2line committed:
improve typography of string_of_title for nobility titles

* Place the ordinal first out of parenthesis followed by a non-breaking space
* Better date span format using endash without surrounding spaces
* Use full roman instead of italic to avoid readibility problems between l/I/1/7

Also try in individu.txt to render the same with “new” splited variables but the
hyperlinks when places are concatenated are not as good as with string_of_title.

a2line committed:
add cancel_links support in geneweb_link

Halbaroth committed:
Fix `Title.select_title_place`

The filter function within the `Title.select_title_place` select
function had a bug. This commit restores the previous behavior.

a2line committed:
titleDisplay format

use column translations, remove italic on dates

a2line commented:
Taking same example as previously shown `http://localhost:2317/grimaldi_w?pz=rainier+louis+henri+maxence+bertrand&nz=grimaldi&m=TT&sm=S&t=prince&p=de+Monaco`

With cid 91f97a5 fix, the list of individuals is now back but there is still a permutation somewhere between the title itself and places internaly because when clicking « de Monaco » in the title — we can click both parts in the HTML header — it lists the various title available on the page of the place.

`http://localhost:2318/grimaldi_w?pz=rainier+louis+henri+maxence+bertrand&nz=grimaldi&m=TT&sm=S&a=A&p=de+Monaco`

![image](https://github.com/user-attachments/assets/5f7ab9e5-7298-458d-9327-88c0fb6d3514)


The parameter `&p=de+Monaco` in URL is correct but the title of the page shows “seigneur” (that is the last entry of the list) instead of “de Monaco” and then choosing a nobility title in list, for exemple “prince”, that should go back to 1st URL, we end up with `t=prince&p=seigneur` instead that shows an empty page again.

`http://localhost:2318/grimaldi_w?pz=rainier+louis+henri+maxence+bertrand&nz=grimaldi&m=TT&sm=S&t=prince&p=seigneur`

a2line commented:
I add that doing the contrary is correct: clicking on `prince` hyperlink in the header, then picking `de Monaco` in the list, we go back to the starting URL.
![image](https://github.com/user-attachments/assets/f13a1377-b088-47f3-b8e1-2cd489b34b7a)

Halbaroth committed:
Do not return clean argument in select functions of `Title` module

Select functions in `Title` module previously returned a cleaned version of
their input. For instance, querying for `de monaco` with `select_title`
would produce `de Monaco` as cleaned version of the input, assuming such
a title exists in your base. This commit removed the feature for the
following reasons:
1. The cleaned version of the input was arbitrarily selected from the
   result list. Before #2287, it was the last element of this list.
2. The implementation of this feature mixes view logic with model logic.
3. Users expect their original input as header of the result list.

Halbaroth commented:
Good catch, I suggest a fix in a new commit. Let me know if you agree with it ;)

hgouraud committed:
wrong closing list tag in print_all_with_place_list

a2line commented:
Thank you @Halbaroth. It's now exactly previous behaviour.

Henri found a bad closing tag `<li>` in “all the title” link. I also have a little work on various typographic problems that I add behind your PR so that Henri can test it too.

I observe a last historical problem when clicking on a title (`prince`) to have the list of its places, then choosing a place (`… de Monaco`), we go back to an URL that keep the extra `&a=A` removing the hyperlinks in the header. I'm looking how to remove that extra parameter right now.

a2line committed:
propagate &a=A only if p is empty
